sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1668,6 +1661,9 @@ public final void run() {
                 // * https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#nanoTime()
                 // * https://github.com/netty/netty/issues/3705
                 if (expiryTime - currentNanoTime <= 0) {
+                    if (logger.isDebugEnabled()) {
```

Comment:
this can become very noisy when requests are added here due to high concurrency or high volumes. Maybe we can change to log the total cleared count during each run?

Reply:
or trace level

Reply:
this log was mean to replace the one on L1648.  It shouldn't be too noisy since it has to meet the condition expiryTime - currentNanoTime <= 0.  Let me know if you know of scenarios where this can fill up the logs.

Reply:
I think what Annie is meaning is that in line 1648 you would do one log per service endpoint - here it is per channel. What is the benefit of having multiple logs for the same endpoint - why not just adding the count in one log?

I am fine with you logs as well - it is debug level - so, I am not too concerned about logging overhead - and number of channels for an endpoint can be >1 - but won't be super high.